src/test/regress/sql/with.sql

```diff
@@ -1081,5 +1081,44 @@ WITH cte AS (
 	SELECT count(*) c1 FROM d
 ) SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
 
+-- Test that conusmer and producer slices are on single segment in case when
+-- Shared Scan is generated over a node with a General locus
+EXPLAIN (COSTS OFF)
+WITH cte AS (
+	SELECT count(*) c1 FROM (VALUES ( 1, 2 ),( 3, 4 )) v
+)
+SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+WITH cte AS (
+    SELECT count(*) c1 FROM (VALUES ( 1, 2 ),( 3, 4 )) v
+)
+SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+RESET optimizer;
+DROP TABLE d;
```

Comment:
I think it's worth moving this test down to the Segment General test. And also need to update the comment. What about something like this:
```
-- Check if sharing is disabled for a General CTE to avoid deadlock if CTE is
-- executed with coordinator gang and joined with n-gang
```